doc/specs/#605 - Search/spec.md

```diff
@@ -0,0 +1,99 @@
+5. The search dialog should not block terminal's view. 
+
+Conhost already has a module for search. It realizes case sensitive or insensitive exact text match search, and it provides methods to select the found word. However, we want to make search as a shared component between Terminal and Console host. Now search module is part of Conhost, and its dependencies include BufferOut and some other types in ConHost such as SCREEN_INFORMATION. In order to make Search a shared component, we need to remove its dependency on ConHost types. BufferOut is already a shared component, but we need to make sure there is no other Conhost dependency.
+
+We will create a SearchBoxControl Xaml UserControl element. When a search process begins, a SearchBoxControl object will be created and attach to TermControl root grid. In other words, one SearchBox is added for each TermControl. The reasons for this design is:
+
+1. Each TermControl corresponds to an iterative Terminal Window and has a individual text buffer. In phrase 1 we are going to search witin the current terminal text buffer. 
+2. If we put the search box under TerminalApp, then the search can only happen on the current focused Terminal. 
+3. If the community does not like the current design, we can lift SearchBox to a higher level. 
+
+Search process:
+1. Search is performed on a XAML AutoSuggestBox. Once the user click the "Find" icon, we start from the cursor (the current input line on the Terminal) or the current selection, and try to find the exact text in the text buffer. The nearest searched one will be selected. And we set the search start point to the selected text. The next time "Find" button is clicked, the search will start before or after the previous searched text.
+2. The user can choose to search up or down by choosing up arrow or down arrow buttons. A blue border will appear on the chosen button. 
+3. The user can choose to do case sensitive or insensitive match by checking a check box. 
+4. If the user click the "X" button, the search stopped and the search box disappears. In phrase one we do not store any state, but in the future we can consider storing the search history. 
+
+## UI/UX Design
+
+![Sol Design](images/SearchBoxControl.png)
+
+Above is the SearchBoxControl. The two buttons with up/down arrows controls the search direction, a blue border will appear on the current selected direction. The checkbox, if checked, means that we are searching case-sensitivily. 
+
+![Sol Design](images/SearchBoxUpSelected.png)
+
+The search box should be on the top right corner of the Terminal window. If the current tab is splitted into panes, each pane can have a individual searchbox. We need to avoid it blocking too much screen contents. 

```

Comment:
```suggestion
The search box should be on the top right corner of the Terminal window. If the current tab is split into panes, each pane can have a individual searchbox. We need to avoid it blocking too much screen contents. 
```